.github/workflows/generate-js.yml

```diff
@@ -0,0 +1,50 @@
+name: 'Generate JavaScript code and open pull request with it'
+
+on:
+  push:
+    branches-ignore:
+     - 'renovate/*'
+     - 'github-actions/generate-js/*'
+    paths-ignore:
+     - '.github/**'
+     - '**.md'
+     - 'action.yml'
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Setup node
+        uses: actions/setup-node@v3
+        with:
+          node-version: 16
+          cache: npm
+
+      - run: npm install
+
+      - run: npm run prepare
+
+      - name: Get word diff to see changes in dist/index.js
+        id: diff
+        run: |
+          EOF=$(dd if=/dev/urandom bs=15 count=1 status=none | base64)
+          echo "GIT_DIFF<<$EOF" >> $GITHUB_OUTPUT
+          git diff --word-diff=porcelain --word-diff-regex=. dist/ | grep '^[+-]' >> $GITHUB_OUTPUT
```

Comment:
You could probably do a more basic diff and use the result of whether there's a diff as a boolean to determine whether to run the next step or not.

Similar thing: https://github.com/cdklabs/cdk-monitoring-constructs/blob/0d79e94c89dba85e5532a355717c199000505fec/.github/workflows/build.yml#L34